FYI: Linux version using FTDI dongle.

eliotb opened the pull request:
Hi Tisham,

Eliot here - we met at LCA and you gave me a CS5490 breakout.
This is just FYI, not a real pull request at this point.

I've created a version of the notebook that uses ordinary python and FTDI dongle to drive the chip.
Many changes compared to original, and still very much work in progress.

Even if you ignore the bulk of it, you might want to look at the byte ordering for R/W 
You use >L and >BBB for struct.unpack, struct.pack,   I think it needs to be <L

eliotb committed:
Linux version using FTDI dongle.

Many changes compared to original
- ? reversed byte packing order for read and write

eliotb committed:
Extend register info dict

Read all documented (in this file) registers.
Remove duplicate reads elsewere.

whatnick commented:
Let us keep this as a "CPython" version of the notebook as opposed to uPython version. Possibly a branch or a fork. Awesome job following through double checking the work.

whatnick commented:
The docs suggest the >L should read/write in big-endian which is what I chose to use and got reasonable output. Cannot really tell without the to float conversions you have added. Temperatures look great over there. Let me retest in micropython.

Reference: https://docs.python.org/2/library/struct.html#byte-order-size-and-alignment

eliotb committed:
Incorporate sign and frac bit count data into register 'address'

Add correct data for all registers

eliotb commented:
I came to my conclusions for the endian struct packing by comparing values read back to the default values given in the datasheet.  Also my V and I readback values were going crazy, but this makes sense if the LSB noise was in the MSB!

In my debug read output default is in []  E.g.

Read    config0 = 0xc02000 [0xc02000]

AFAIKS the datasheet is not explicity about what the serial byte order is!

eliotb committed:
CS5490 driver as regular python module

eliotb committed:
Enable energy pulse output.

Pulsewidth about 16ms

eliotb committed:
Get baudrate switch working, rework reset

Baud switch at startup seems to work ok.
Still something a bit funky about reset behaviour?  Maybe serial
port baud change doesn't work reliably?
